internal/services/serviceconnector/helper.go

```diff
@@ -270,8 +277,42 @@ func flattenTargetService(input servicelinker.TargetServiceBase) string {
 	if value, ok := input.(servicelinker.AzureResource); ok {
 		if value.Id != nil {
 			targetServiceId = *value.Id
+			if parsedId, err := validateStorageBlob(targetServiceId); err == nil {
+				targetServiceId = parsedId.ID()
+			}
 		}
 	}
 
 	return targetServiceId
 }
+
+func validateStorageBlob(input string) (*StorageBlobId, error) {
```

Comment:
shouldn'y an ID + vlaidators for a storage blob not already exist?

Reply:
Hi @katbyte , the existing validator is not for this default storage blob comes with the account. Using the existing one will cause error because the format of the resource id is different.

Reply:
👍  - could we generate it in resourcesids.go? ie like:
```
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=WebApp -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.Web/sites/site1
```

Reply:
Sure. Replaced with generated code.